File: CS/Classes/DirectoryHelper.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Views.Grid;
using System.IO;

namespace WindowsApplication1
{
    public static class DirectoryHelper
    {
        public static List<GridFile> GetFiles(string path)
        {
            string[] files = Directory.GetFiles(path);
            List<GridFile> res = new List<GridFile>();
            for (int i = 0; i < files.Length; i++)
            {
                res.Add(new GridFile(files[i], i));
            }
            return res;
        }
    }
}

File: VB/Classes/DirectoryHelper.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Views.Grid
Imports System.IO

Namespace WindowsApplication1
	Public NotInheritable Class DirectoryHelper
		Private Sub New()
		End Sub
		Public Shared Function GetFiles(ByVal path As String) As List(Of GridFile)
			Dim files() As String = Directory.GetFiles(path)
			Dim res As New List(Of GridFile)()
			For i As Integer = 0 To files.Length - 1
				res.Add(New GridFile(files(i), i))
			Next i
			Return res
		End Function
	End Class
End Namespace
